3_harmonize/src/harmonize_sdd.R

```diff
@@ -13,483 +9,997 @@ harmonize_sdd <- function(raw_sdd, p_codes,
-    inner_join(x = .,
+
+  unit_table_out_path <- "3_harmonize/out/sdd_unit_table.csv"
+
+  write_csv(unit_conversion_table, unit_table_out_path)
+
+  converted_units_sdd <- sdd_no_na %>%
+    left_join(x = .,
                y = unit_conversion_table,
-               by = "ResultMeasure.MeasureUnitCode") %>%
-    mutate(harmonized_value = ResultMeasureValue * conversion,
-           harmonized_units = "m") 
-  
+               by = "harmonized_units") %>%
+    mutate(harmonized_value = as.numeric(harmonized_value) * conversion,
+           harmonized_units = "m") %>%
+    # filter for successful converted units and bottoms
+    filter(!is.na(harmonized_value))
+
+    # Plot and unit codes that didn't make through joining
+    sdd_no_na %>%
+      anti_join(x = .,
+                y = converted_units_sdd,
+                by = "ResultMeasure.MeasureUnitCode") %>%
+      count(ResultMeasure.MeasureUnitCode, name = "record_count") %>%
+      plot_unit_pie() %>%
+      ggsave(filename = "3_harmonize/out/sdd_unit_drop_pie.png",
+            plot = .,
+            width = 6, height = 6, units = "in", device = "png")
+
   # How many records removed due to unit harmonization?
   print(
     paste0(
-      "Rows removed while converting to numeric values: ",
-      nrow(sdd_harmonized_values) - nrow(converted_units_sdd)
+      "Percentage of rows removed while converting units: ", #***
+      nrow(sdd_no_na) - nrow(converted_units_sdd)
     )
   )
-  
+
   dropped_harmonization <- tibble(
-    step = "sdd harmonization",
+    step = "sdd unit conversion",
     reason = "Dropped rows while harmonizing units",
     short_reason = "Harmonize units",
-    number_dropped = nrow(sdd_harmonized_values) - nrow(converted_units_sdd),
+    number_dropped = nrow(sdd_no_na) - nrow(converted_units_sdd),
     n_rows = nrow(converted_units_sdd),
-    order = 6
+    order = 8
   )
-  
-  
-  # Aggregate analytical methods --------------------------------------------
-  
+
+  # Clean and flag bottom depth data -------------------------------------------
+
+  sdd_bottom_depth <- converted_units_sdd %>%
+    # filter for samples that have bottom text in the comment columns and have no information in the bottom depth columns.
+    filter(bottom_tag == 1 & negate_bottom_tag != 1 &
+             (is.na(ActivityBottomDepthHeightMeasure.MeasureValue)&is.na(ActivityBottomDepthHeightMeasure.MeasureUnitCode)))
+
+  # Keep important data
+  sdd_bottom_depth <- sdd_bottom_depth %>%
+    select(
+      index, harmonized_value, harmonized_units,
+      ActivityBottomDepthHeightMeasure.MeasureValue,
+      ActivityBottomDepthHeightMeasure.MeasureUnitCode,
+      bottom_tag, negate_bottom_tag
+    )
+
+  print(
+    paste(
+      round((nrow(sdd_bottom_depth)) / nrow(converted_units_sdd) * 100, 3),
+      "% of samples have no reported ActivityBottomDepthHeightMeasure.MeasureValue that can be 
+      back filled with harmonized values where Secchi on bottom was indicated."
```

Comment:
Suggestion: Could this print statement be clarified a little? I'm not entirely sure I understand the meaning. Perhaps:
```suggestion
      "% of samples have no reported ActivityBottomDepthHeightMeasure.MeasureValue.  These will be 
      back filled with harmonized values where Secchi on bottom was indicated."
```